### PR TITLE
Error message

### PR DESCRIPTION
**Pull request by alecrabbit**
Hi!
during development  of my own project [php-dev-helper](https://github.com/alecrabbit/sh-php-dev-helper) 
I was annoyed with exception trace of absent `.multi-tester.yml` or when it was empty
so I've added - nice error messages

### CODE REVIEW

File: src/MultiTester/Summary.php [hunk 1]
```diff
@@ -45,32 +43,46 @@ function_exists('posix_isatty') &&
...
-        $failureFinalString = '%d / %d     %s broken by current changes.';
-        $passed = 0;
-
-        if ($this->isColored()) {
-            $successString = "\033[42;97m $successString \033[0m";
-            $failureString = "\033[41;97m %s    Failure \033[0m";
-            $successFinalString = "\033[42;97m %d / %d     No project broken by current changes. \033[0m";
-            $failureFinalString = "\033[41;97m %d / %d     %s broken by current changes. \033[0m";
-        }
+        if ($count > 0) {
```
Comment (kylekatarnls) [inline src/MultiTester/Summary.php]: Please prefer early-return over else (according to PHPMD rules):
```php
if ($count === 0) {
    return 'No project(s) tested, empty config file?' . PHP_EOL;
}
```
It avoid to indent the whole function with 4 extra spaces.

File: src/MultiTester/Summary.php [hunk 2]
```diff
@@ -45,32 +43,46 @@ function_exists('posix_isatty') &&
...
-        $output .= "\n" . sprintf($passed === $count ? $successFinalString : $failureFinalString, $passed, $count, ($count - $passed) . ' project' . ($count - $passed > 1 ? 's' : '')) . "\n";
+            foreach ($this->state as $package => $success) {
+                $passed += $success ? 1 : 0;
+                $output .= sprintf($success ? $successString : $failureString, str_pad($package, $pad)) . "\n";
+            }
 
+            $output .= "\n" . sprintf($passed === $count ? $successFinalString : $failureFinalString, $passed, $count,
+                    ($count - $passed) . ' project' . ($count - $passed > 1 ? 's' : '')) . "\n";
+        } else {
+            $output = 'No project(s) tested, empty config file?' . PHP_EOL;
```
Comment (kylekatarnls) [inline src/MultiTester/Summary.php]: This line is not covered by unit tests. Can you consider adding a unit test with an empty config that verifies you get the errors.

Typo: `No projects` = `0 projects` and in English, 0 is plural so you don't need the parentheses.

Please use `"\n"` instead of `PHP_EOL` for consistency. If for some reason we want to support other end-of-lines, it should be changed everywhere.

And maybe we can have a more helpful message: `check you have projects in your config file: ' . $this->getMultiTesterFile()`

**Review by kylekatarnls — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution. Please review my suggestions.

File: bin/multi-tester [hunk 3]
```diff
@@ -43,4 +44,15 @@ if (!getLoader()) {
...
+} catch (\Exception $e) {
+    if (Summary::isColorSupported()) { // is colored?
+        $errorString = " \033[41;1;97m  ERROR  \033[0m\033[31m {$e->getMessage()} \033[0m" ;
+    } else {
+        $errorString = 'ERROR:' . $e->getMessage();
+    }
+    echo PHP_EOL . $errorString . PHP_EOL;
+    exit(1);
+}
+exit($code);
```
Comment (kylekatarnls) [inline bin/multi-tester:88]: What is annoying for you can be useful for other people. Stack trace exists because it can help to debug an error. It should remain visible at least when the `--verbose` option is in use. And if the stack trace is hidden when the output is not verbose, errors should be followed by a tip `Use --verbose option to get more details.`

**Review by kylekatarnls — CHANGES_REQUESTED** [review]

File: bin/multi-tester [hunk 4]
```diff
@@ -43,4 +60,29 @@ if (!getLoader()) {
     exit(1);
 }
 
-exit((new MultiTester())->run($argv) ? 0 : 1);
+try {
+    $multiTester = new MultiTester();
+    $code = $multiTester->run($argv) ? 0 : 1;
+} catch (ZeroProjectsTestedException $e) {
+    showError($e);
```
Comment (kylekatarnls) [inline bin/multi-tester:68]: The same way you check `$multiTester->isVerbose()` then you could test `$multiTester->isColored()` so the static method is no longer needed and color could be configurable for errors too.

Comment (alecrabbit) [reply bin/multi-tester:68]: be patient )) optimizations will follow

Comment (alecrabbit) [reply bin/multi-tester:68]: I'll let you know when PR is ready

Comment (alecrabbit) [reply bin/multi-tester:68]: also it seems to me that static method is needed in case when config file is absent

**Comment by alecrabbit** [comment]
I suggest accept this PR as is and bump version to 1.3

File: src/MultiTester/Summary.php (hunk 2 above)
Comment (alecrabbit) [reply src/MultiTester/Summary.php]: I've not yet figured out how to cover that line, so I've just ignored it for now

**Comment by alecrabbit** [comment]
added test

**Comment by alecrabbit** [comment]
@kylekatarnls Kyle, now PR is ready )
